### PR TITLE
Hello rails react app

### PR DESCRIPTION
**Pull request by Roland-Ntwali**
In This PR, I did the following:

- Create a new Rails app called 'hello-rails-react'.
- Use the command rails new -j webpack my_new_rails_app as described in the previous lesson's tutorial.
- Verify that the jsbundling-rails gem is added to your Gemfile.
- Initialize project with Git.
- Install needed node dependencies as described in the previous lesson's tutorial.
- Create a babel.config.js in the root folder of your project as described in the previous lesson's tutorial.
- Update webpack.config.js to include the babel-loader as described in the previous lesson's tutorial.
- project has a Postgres database set up.
- Create a table for storing your messages and create 5 different greetings.
- Create an API endpoint that selects a random greeting from your table (you will need a controller with an action and Rails route).
- Create a static view that will be the root of your app
- Create the App component with react-router and render it in your static view.
- Create the Greeting component that will display a greeting. Set it up as a route in your App component.
- Create a store, an action, and a reducer that will connect to your API endpoint to get the random greeting.
- Display the random greeting in your Greeting component.

### CODE REVIEW

File: app/javascript/application.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import App from './components/App';
+
+// function App() {
+//   return (<h1>Hello World!</h1>);
+// }
+
+// ReactDOM.render(
+//   <App/>,
+//   document.getElementById('root'),
+// );
```
Comment (julie-ify) [inline app/javascript/application.js]: - [ ] Kindly remove commented lines of code. They looky messy and make your code hard to read.

File: app/javascript/components/Greeting.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import React, { useEffect } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { getGreetings } from "../controllers/greetingsSlice";
```
Comment (julie-ify) [inline app/javascript/components/Greeting.js]: - [ ] There is an import error here. You should import from `../redux/greetingsSlice`

File: app/javascript/components/App.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import React from "react";
+import { BrowserRouter as Router, Route, Routes } from "react-router-dom";
+import { Provider } from "react-redux";
+import Random from "./Greeting";
+import store from "../controllers/store";
```
Comment (julie-ify) [inline app/javascript/components/App.js]: - [ ] Kindly import from `..redux/store`

File: app/javascript/components/App.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+const App = () => (
+  <Provider store={store}>
+    <Router>
+      <Routes>
+        <Route exact path="/" element={<Random />} />
```
Comment (julie-ify) [inline app/javascript/components/App.js:11]: - [ ] When I visit the root route, I see no greeting. Kindly fix it.

 image

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @Roland-Ntwali,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by andersonlebon — APPROVED** [review]
## Hi @Roland-Ntwali 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

nodding-yes


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question `@andersonlebon` so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**